README.md

```diff
@@ -2,147 +2,142 @@
+|             | int4wo-64          | 12.843              |  199.81       | 746.45                  | 3.74             | 4.75            |
+|             | int4wo-64-GPTQ     | 12.489              |  199.81       | 746.45                  | 3.74             | 4.75            |
+| Llama-3-8B  | Base (bfloat16)    | N/A                 |  94.91        | 1424.58                 | 15.01            | 16.43           |
+|             | int8dq             | N/A                 |  8.41         | 63.23                   | 7.52             | 9.24            |
+|             | int8wo             | N/A                 |  136.75       | 1028.38                 | 7.52             | 10.42           |
+|             | int4wo-64          | N/A                 |  179.41       | 757.45                  | 4.22             | 6.88            |
+
+note: Int8 dynamic quantization works best on compute bound models like [SAM](https://github.com/pytorch-labs/segment-anything-fast) whereas Llama with batchsize=1 tends to be memory bound, thus the rather low performance.
+
+And a quick crash course on inference quantization to help parse the above table. Int4 quantization is an ambiguous term because there's the dtype in which a layer is represented and then the dtype in which the computation is done. For example, if you're using Weight-Only (wo) int4 quantization that means that the layer will be upcasted to a larger dtype like fp16 so an int4 matrix multiplication is defined as `F.linear(input, weight.to(input.dtype))` whereas if it's possible to perform the computation using the smaller dtype directly pending support by a hardware vendor then that means you can perform `F.linear(input, weight)` directly and this is what we refer to as Dynamic-Quantization (dq). Naive quantization algorithms are also notoriously sensitive to outliers so we also typically set a group size that applies a scale factor per group of 64 elements in the case of `int4wo64`.
+
+
+#### With intrusive code changes
+
+In some cases we rewrote popular GenAI models to be significantly faster in native PyTorch as in no C++/CUDA to achieve at the time SOTA inference performance. These involve more intrusive code changes.
+
+* 8x speedups for Image segmentation models with [sam-fast](https://pytorch.org/blog/accelerating-generative-ai)
+* 10x speedups for Language models with [gpt-fast](https://pytorch.org/blog/accelerating-generative-ai-2)
+* 3x speedup for Diffusion models with [sd-fast](https://pytorch.org/blog/accelerating-generative-ai-3)
+
+### Training
+
+We've added support for semi-structured 2:4 sparsity with over 30% speedups on ViT-L
+
+The code change is a 1 liner with the full example available [here](torchao/sparsity/training/)
+
+
+```python
+swap_linear_with_semi_sparse_linear(model, {"seq.0": SemiSparseLinear})
+```
+
+For VIT-L MLP shapes on a NVIDIA A100 we see the following results:
+
+|                     |   act24   |   dense   |   w24    | s24_inp_sparsify24 | s24_inp_clone |
```

Comment:
discussed this with @jcaip offline but would be good to make the columns here more clear.

Reply:
@jcaip please send me a clearer table when you have it - keep in mind I want to make some claim about end to end performance

Reply:
Yes, I'm making some changes to the benchmarking script for e2e ViT. Will send something by EOD.

Reply:
Btw depending on what time you're done feel free to just push changes to my PR directly and merge since I'll be unavailable tomorrow